### PR TITLE
Changes for Exposing RamlDiffService using Spark

### PR DESCRIPTION
**Pull request by harishputhran**

### CODE REVIEW

File: src/main/java/service/RamlDiffService.java [hunk 1]
```diff
@@ -67,10 +78,35 @@ public void diff(String later, String older) throws Exception {
...
     new RamlDiffService().diff("src/test/resources/github.raml", "src/test/resources/github-api-v3.raml");
   }
-
+*/
+  
+  public static void main(String[] args) throws Exception {
+    get(RAML_DIFF_SERVICE, (request, response) -> { 
+        return RAML_DIFF_SERVICE_AVAILABLE_TEXT;    
+    });
+    post(FIND_ALL_DIFFERENCES_CONTEXT, (request, response) -> {
```
Comment (araman-io) [inline src/main/java/service/RamlDiffService.java]: Why is a post and not a get. I can see that you are taking a JSON of some form as the post body. However the POST in REST specification has a specific meaning. Wouldnt it be nice to just treat this as query params ?

File: src/main/java/service/RamlDiffService.java [hunk 2]
```diff
@@ -67,10 +78,35 @@ public void diff(String later, String older) throws Exception {
...
+      String oldRamlFilePath = requestObj.getOldRamlFileURL();
+      String newRamlFilePath = requestObj.getNewRamlFileURL();
+      
+      URL oldFileUrl = new URL(oldRamlFilePath);
+      URL newFileUrl = new URL(newRamlFilePath);
+      
+      List<ActionDiff> allDifferences = new RamlDiffService().diff(newFileUrl.getFile(), oldFileUrl.getFile());
+      
+      RamlDiffServiceResponse responseObj = new RamlDiffServiceResponse();
+      responseObj.setRamlFileDifferences(allDifferences.stream().flatMap(eachDiff -> {
```
Comment (araman-io) [inline src/main/java/service/RamlDiffService.java]: why do we need to convert the action diff into some other response object. Wouldnt it be nice to just stream the ActionDiff objects back as a JSON.

thanks

File: src/main/java/service/RamlDiffService.java (hunk 1 above)
Comment (harishputhran) [inline src/main/java/service/RamlDiffService.java]: Have implemented Get.

File: src/main/java/service/RamlDiffService.java (hunk 2 above)
Comment (harishputhran) [inline src/main/java/service/RamlDiffService.java]: Added Response Object to capture any error messages in Raml Difference finders. I had not written that piece of code. Have removed the Request & Response beans as well as JSON Utility.

**Pull request merged**